lib/OpenQA/Worker/Jobs.pm

```diff
@@ -153,16 +154,44 @@ sub upload {
     close $log;
     printf "uploading %s\n", $filename if $verbose;
 
-    my $ua_url = $OpenQA::Worker::Common::url->clone;
-    $ua_url->path("jobs/$job_id/artefact");
+    my $retry_counter = 5;
+    my $retry_limit   = 5;
+    my $tics          = 5;
+    my $res;
+
+    while ($retry_counter) {
+        my $ua_url = $OpenQA::Worker::Common::url->clone;
+        $ua_url->path("jobs/$job_id/artefact");
+
+        my $tx = $OpenQA::Worker::Common::ua->build_tx(POST => $ua_url => form => $form);
+        # override the default boundary calculation - it reads whole file
+        # and it can cause various timeouts
+        my $headers = $tx->req->headers;
+        $headers->content_type($headers->content_type . "; boundary=$boundary");
 
-    my $tx = $OpenQA::Worker::Common::ua->build_tx(POST => $ua_url => form => $form);
-    # override the default boundary calculation - it reads whole file
-    # and it can cause various timeouts
-    my $headers = $tx->req->headers;
-    $headers->content_type($headers->content_type . "; boundary=$boundary");
+        $res = $OpenQA::Worker::Common::ua->start($tx);
 
-    my $res = $OpenQA::Worker::Common::ua->start($tx);
+        # Upload known server failures (Instead of anything that's not 200)
+        if ($res->res->is_status_class(500)) {
+
+            if ($retry_counter) {
```

Comment:
@coolo, actually there was also a problem with my previous code (There were 4 uploads instead of 5). New code should fix it now.

Had to add `$regular_upload_failed` so that the warning woudn't show up on the first attempt along the wait thingie...